modules/msr/main.tf

```diff
@@ -64,26 +66,28 @@ resource "aws_lb" "msr" {
 }
 
 resource "aws_lb_target_group" "mke_msr_api" {
+  count    = var.msr_count >= 1 ? 1 : 0
   name     = "${var.cluster_name}-msr-api"
   port     = 443
   protocol = "TCP"
   vpc_id   = var.vpc_id
 }
 
 resource "aws_lb_listener" "mke_msr_api" {
-  load_balancer_arn = aws_lb.msr.arn
+  count             = var.msr_count >= 1 ? 1 : 0
+  load_balancer_arn = aws_lb.msr[0].arn
   port              = 443
   protocol          = "TCP"
 
   default_action {
-    target_group_arn = aws_lb_target_group.mke_msr_api.arn
+    target_group_arn = aws_lb_target_group.mke_msr_api[0].arn
     type             = "forward"
```

Comment:
Shouldn't you use the variable `count` instead of using 0th index here?

Reply:
fixed